polynote-kernel/src/main/scala/polynote/kernel/PolyKernel.scala

```diff
@@ -70,37 +60,51 @@ class PolyKernel private[kernel] (
   // TODO: duplicates logic from runCell
   private def runPredef(interp: LanguageInterpreter[IO], language: String): IO[Stream[IO, Result]] =
     interp.init() *> {
-      interp.predefCode match {
-        case Some(code) => taskManager.runTaskIO(s"Predef $language", s"Predef ($language)")(_ => interp.runCode(-1, Nil, Nil, code)).map {
-          results => results.collect {
-            case v: ResultValue => v
-          }.evalTap {
-            v => symbolTable.publishAll(symbolTable.RuntimeValue.fromResultValue(v, interp).toList)
+      val index = notebookContext.first.map(_.id).getOrElse(0.toShort) - 1
```

Comment:
1. maybe this should be `id` rather than `index`, since this is actually setting the CellId and not necessarily the index. 
2. since this is setting the cell id, any reason it shouldn't always be `-1`? If someone added/removed cells prior to running the predef I think it's possible that `notebookContext.first` would have an ID that's not necessarily the smallest (for example, if I insert a cell at the top of my notebook before running the predef, I think `notebookCell.first.id` would be the number of cells in the notebook which would cause a collision where `predef.id == lastCell.id`. (unless I'm missing some other change?)

Reply:
Ohh, I see - because we run the Scala interpreter every time there could be multiple Predefs being run? which is why you can't just hard-code it to -1... but I think the problem with the notebook state being changed by the user still stands unfortunately.

Reply:
I guess maybe a check if `id <= 0` is all you need to solve that tho so not too bad

Reply:
Yep, you got it – there could be multiple predefs (one per interpreter). In #191 I added some logic to ensure that it always ends up < 0. Also added a comment explaining.